app/lib/features/chat_ng/providers/notifiers/chat_room_messages_notifier.dart

```diff
@@ -0,0 +1,202 @@
+import 'dart:async';
+
+import 'package:acter/common/extensions/options.dart';
+import 'package:acter/features/chat_ng/models/chat_room_state/chat_room_state.dart';
+import 'package:acter/features/chat/providers/chat_providers.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/widgets.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:logging/logging.dart';
+import 'package:riverpod/riverpod.dart';
+
+final _log = Logger('a3::chat::room_notifier');
+
+class ChatRoomMessagesNotifier extends StateNotifier<ChatRoomState> {
+  final Ref ref;
+  final String roomId;
+  late TimelineStream timeline;
+  late Stream<RoomMessageDiff> _listener;
+  late StreamSubscription<RoomMessageDiff> _poller;
+
```

Comment:
updated for the diffing function, this acts almost the same as the previous implementation